lib/auth/permissions.go

```diff
@@ -27,8 +27,8 @@ import (
 )
 
 // NewRoleAuthorizer authorizes everyone as predefined role
-func NewRoleAuthorizer(r teleport.Role) (Authorizer, error) {
-	authContext, err := contextForBuiltinRole(r)
+func NewRoleAuthorizer(recordAtProxy bool, r teleport.Role) (Authorizer, error) {
```

Comment:
this is not scalable, I would assume that proxy will switch to different role name based on whether it is recording or not, why does authorizer has to know about this flag?

Reply:
Ideally the proxy would have it's own role, but to access the Auth Server first it has to register itself with the Auth Server with the role it will assume, which it can't do because the Auth Server stores the role it will assume in `services.ClusterConfig`.

So we have two choices, either we make the authorizer smarter (what I did in this PR) or we create another role that can just be used to query the Auth Server for `services.ClusterConfig`.

I can change the behavior here so it takes a `services.ClusterConfiguration` that way it can change it's value on the fly.

Reply:
@klizhentas As per our discussion, I've backed out these changes. I'll address this in another PR since this one has gotten too big already: https://github.com/gravitational/teleport/issues/1431